core/parallel_state_processor.go

```diff
@@ -315,9 +315,11 @@ func (p *ParallelStateProcessor) Process(block *types.Block, statedb *state.Stat
 
 		cleansdb := statedb.Copy()
 
-		if len(header.TxDependency) > 0 {
-			shouldDelayFeeCal = delayMap[i]
+		if msg.From() == coinbase {
+			shouldDelayFeeCal = false
+		}
 
+		if len(header.TxDependency) > 0 {
```

Comment:
Should we check if `len(header.TxDependency) == len(tasks)`? If lengths mismatch, we shouldn't use any dependency data.

Reply:
Nice idea. Thanks!